build/parseSpec.c

```diff
@@ -100,6 +101,30 @@ int isPart(const char *line)
     return (p->token ? p->part : PART_NONE);
 }
 
+static const char * partName(int part)
+{
+    const struct PartRec *p;
+
+    for (p = partList; p->token != NULL; p++) {
+	if (p->part == part) {
+	    return p->token;
+	}
+    }
+    return NULL;
+}
+
+static const int partBeforeBuildOnly(int part)
+{
+    const struct PartRec *p;
+
+    for (p = partList; p->token != NULL; p++) {
+	if (p->part == part) {
+	    return p->beforebuildonly;
+	}
+    }
+    return 0;
```

Comment:
I'd combine these two into findPart() that returns a pointer to the struct, whose values you can then just use directly in the caller.

Reply:
There's still this one :innocent:

Reply:
As I wrote above this is not quite as simple as it looks and I ma not that interested in refactoring the whole parting code. Yes, it looks wasteful and over complicated but I'd rather not change  a whole bunch of functions just to pass a new pointer around.

Reply:
Oh, I see you mentioned this below. isPart() is a different matter, but .. 

Once you have 

```
static const struct PartRec * getPart(int part);
```

...the check becomes something like:

```
        if (stage == PARSE_GENERATED) {
            const struct PartRec *p = getPart(parsePart);
            if (p && p->prebuildonly) {
                rpmlog(RPMLOG_ERR,
                        _("Section %s is not allowed after build is done!\n"),
                        p->token);
                goto errxit;
            }
        }   
```

...and at that point you realize this whole if belongs into a separate helper function that checks whether the tag was legitimate for use here. parseSpecSection() is big enough as it is.

Reply:
Not sure if the new code is that more readable. But whatever...